src/Microsoft.DotNet.XHarness.Android/AdbRunner.cs

```diff
@@ -424,19 +448,27 @@ public List<string> PullFiles(string apkPackageName, string devicePath, string l
         }
     }
 
-    public Dictionary<string, string?> GetAttachedDevicesWithProperties(string property)
+    public List<AndroidDevice> GetAttachedDevicesWithProperties()

```

Comment:
Couldn't it be just called `GetAttachedDevices()`? Also I think it's a good practice to use immutable collections (like `IReadOnlyList<AndroidDevice>`) where possible unless we expect the called to modify the result. I'm not sure if it applies in this case.

Reply:
Yup, makes sense.